README.md

```diff
@@ -7,6 +7,202 @@
 [![Coverage Status](https://coveralls.io/repos/github/voxpupuli/puppet-webhook/badge.svg?branch=master)](https://coveralls.io/github/voxpupuli/puppet-webhook?branch=master)
 [![Dependency Status](https://gemnasium.com/badges/github.com/voxpupuli/puppet-webhook.svg)](https://gemnasium.com/github.com/voxpupuli/puppet-webhook)
 
-The puppet webhook server is a Sinatra-based application for receiving webhooks that will trigger r10k runs on your puppet master.
+## What is puppet_webhook
 
-**NOTE: This is very much still in development and NOT stable**
+puppet_webhook is a Sinatra-based application receiving REST-based calls to trigger Puppet and r10k-related tasks such as:
+
+* Webhooks from Source Code systems to trigger r10k environment and module deploys
+* REST calls from systems to trigger Puppet Decommissions such as:
+    * `puppet node clean`
+    * `puppet cert clean`
+    * `puppet cert revoke`
+    * etc.
+* Send notifications via Slack
+
+## Prerequisites
+
+* Ruby 2.1.9 or greater
+* Puppet 4.7.0 or greater
+* r10k gem
+* MCollective and MCollective-r10k (Only for multi-master syncronization)
+    * Currently Mcollective-r10k is only available from [puppet-r10k](https://github.com/voxpupuli/puppet-r10k)
+
+## Installation
+
+Currently the only supported installation method is via RubyGems. 
+
+`gem install puppet_webhook`
+
+RPM and DEB packages are planned for future releases.
```

Comment:
side node: I can provide archlinux packages if you like, as soon as a gem is published.